### ISSUE TITLE
single field avro schemas being incorrectly unwrapped?

### ISSUE THREAD

**Issue by tonyclyde** [AbsaOSS/ABRiS#58]
I have an avro schema that consists of a single field (not my choice, someone elses that I need to publish to) and I think it may be getting incorrectly "unwrapped" to a bare `string` field here:  https://github.com/AbsaOSS/ABRiS/blob/master/src/main/scala/za/co/absa/abris/avro/sql/SchemaProvider.scala#L74

This gives me an error:

```
org.apache.spark.sql.avro.IncompatibleSchemaException: Cannot convert Catalyst type StructType(StructField(some_field,StringType,false)) to Avro type "string".
at org.apache.spark.sql.avro.AvroSerializer.newStructConverter(AvroSerializer.scala:205)
```

```
{
  "type": "record",
  "name": "key",
  "fields": [
    {
      "name": "some_field",
      "type": "string"
    }
  ]
}
```

but this seems to work:

```
{
  "type": "record",
  "name": "wrapper",
  "fields": [
    {
      "name": "wrapped",
      "type": 
        {
          "type": "record",
          "name": "key",
          "fields": [
            {
              "name": "some_field",
              "type": "string"
            }
          ]
        }
      }
    ]
}
```

**Comment by cerveada** [comment]
Yes, this is a bug, thanks for reporting.

**Comment by cerveada** [comment]
Fixed in 3.0.2
